include/circt/Dialect/FIRRTL/FIRRTLStructure.td

```diff
@@ -253,4 +253,59 @@ def FMemModuleOp : FIRRTLModuleLike<"memmodule"> {
   let hasCustomAssemblyFormat = 1;
 }
 
+def ClassOp : FIRRTLModuleLike<"class", [
+  SingleBlock, NoTerminator,
+  DeclareOpInterfaceMethods<FModuleLike, [
+    // Class Ops do not support port annotations.
+    // Override this method to return an empty array attr.
+    "getPortAnnotationsAttr"]>
+]> {
+  let summary = "FIRRTL Class";
+  let description = [{
+    The "firrtl.class" operation represents a property-only object, including a
```

Comment:
Saying this represents a property-only object is slightly glossing over the relationship between class and object to me. Maybe this could say "represents the definition of a property-only object", or something like that?